proof-producer/bin/proof-producer/CMakeLists.txt

```diff
@@ -75,14 +83,32 @@ function(setup_proof_generator_target)
         Boost::program_options
         Boost::thread
     )
+
+    target_link_libraries(${ARG_TARGET_NAME}-lib INTERFACE
+        crypto3::all
+
+        crypto3::transpiler
+
+        Boost::filesystem
+        Boost::log
+        Boost::program_options
+        Boost::thread
+    )
 endfunction()
 
 # Declare single-threaded target
 set(SINGLE_THREADED_TARGET "${CURRENT_PROJECT_NAME}-single-threaded")
 setup_proof_generator_target(TARGET_NAME ${SINGLE_THREADED_TARGET})
 
+
 # Declare multi-threaded target
 set(MULTI_THREADED_TARGET "${CURRENT_PROJECT_NAME}-multi-threaded")
```

Comment:
Out-of-scope question: do we really need two different binaries for single- and multi-threaded versions of proof-producer? As I see, they don't even have compile-time definitions, so probably it would be easy to combine them into one and add cmd option for threads. Not mentioning that single-threaded version are probably not going to be used in production.

Reply:
`multi-threaded` uses [actor_zk](https://github.com/NilFoundation/placeholder/blob/1d4fd9ece5040756890083a1cb33c2618f1b47e5/proof-producer/bin/proof-producer/CMakeLists.txt#L106) from `parallel-crypto3`.
As result the same functions in `single-threaded `and `multi-threaded` binaries will have different implementations.
Generally it's possible to combine into one binary and split implementation by using command line option, but not sure if it make sense, And defiantly not in scope this PR)